x-pack/plugins/security/server/authentication/api_keys.test.ts

```diff
@@ -4,57 +4,110 @@
 
-describe('createAPIKey()', () => {
-  it('returns null when security feature is disabled', async () => {
-    const result = await createAPIKey({
-      body: {
+  describe('create()', () => {
+    it('returns null when security feature is disabled', async () => {
+      mockIsSecurityFeatureDisabled.mockReturnValue(true);
+      const result = await apiKeys.create(httpServerMock.createKibanaRequest(), {
         name: '',
         role_descriptors: {},
-      },
-      loggers: loggingServiceMock.create(),
-      callAsCurrentUser: mockCallAsCurrentUser,
-      isSecurityFeatureDisabled: () => true,
+      });
+      expect(result).toBeNull();
+      expect(mockScopedClusterClient.callAsCurrentUser).not.toHaveBeenCalled();
     });
-    expect(result).toBeNull();
-    expect(mockCallAsCurrentUser).not.toHaveBeenCalled();
-  });
 
-  it('calls callCluster with proper body arguments', async () => {
-    mockCallAsCurrentUser.mockResolvedValueOnce({
-      id: '123',
-      name: 'key-name',
-      expiration: '1d',
-      api_key: 'abc123',
-    });
-    const result = await createAPIKey({
-      body: {
+    it('calls callCluster with proper body arguments', async () => {
+      mockIsSecurityFeatureDisabled.mockReturnValue(false);
+      mockScopedClusterClient.callAsCurrentUser.mockResolvedValueOnce({
+        id: '123',
         name: 'key-name',
-        role_descriptors: { foo: true },
         expiration: '1d',
-      },
-      loggers: loggingServiceMock.create(),
-      callAsCurrentUser: mockCallAsCurrentUser,
-      isSecurityFeatureDisabled: () => false,
-    });
-    expect(result).toEqual({
-      api_key: 'abc123',
-      expiration: '1d',
-      id: '123',
-      name: 'key-name',
-    });
-    expect(mockCallAsCurrentUser).toHaveBeenCalledWith('shield.createAPIKey', {
-      body: {
+        api_key: 'abc123',
+      });
+      const result = await apiKeys.create(httpServerMock.createKibanaRequest(), {
         name: 'key-name',
         role_descriptors: { foo: true },
         expiration: '1d',
-      },
+      });
+      expect(result).toEqual({
+        api_key: 'abc123',
+        expiration: '1d',
+        id: '123',
+        name: 'key-name',
+      });
+      expect(mockScopedClusterClient.callAsCurrentUser).toHaveBeenCalledWith(
+        'shield.createAPIKey',
+        {
+          body: {
+            name: 'key-name',
+            role_descriptors: { foo: true },
+            expiration: '1d',
+          },
+        }
+      );
+    });
+  });
+
+  describe('invalidate()', () => {
+    it('returns null when security feature is disabled', async () => {
+      mockIsSecurityFeatureDisabled.mockReturnValue(true);
+      const result = await apiKeys.invalidate(httpServerMock.createKibanaRequest(), {
+        id: '123',
+      });
+      expect(result).toBeNull();
+      expect(mockScopedClusterClient.callAsCurrentUser).not.toHaveBeenCalled();
+    });
+
+    it('calls callCluster with proper body arguments', async () => {
```

Comment:
__nit:__ would you mind adding a simple test that verifies we send only `id` as ES request body param and skip anything else that may be passed (e.g. from JavaScript consumer)?